test/support/test_helper.ex

```diff
@@ -45,4 +45,18 @@ defmodule Xandra.TestHelper do
       Process.sleep(@sleep_interval)
       wait_for_passing(time_left - @sleep_interval, fun)
   end
+
+  def mirror_telemetry_event(event_name) do
```

Comment:
You can use [telemetry_test](https://hexdocs.pm/telemetry/1.2.1/telemetry_test.html) for this, I’m using it already in some of the code I pushed yesterday and this morning.